client/main.lua

```diff
@@ -466,77 +468,72 @@ function Hotwire(vehicle, plate)
 end
 
 function CarjackVehicle(target)
-    isCarjacking = true
-    canCarjack = false
-
-    loadAnimDict('mp_am_hold_up')
-
-    local vehicle = GetVehiclePedIsUsing(target)
-    local occupants = GetPedsInVehicle(vehicle)
-    for p=1,#occupants do
-        local ped = occupants[p]
-        CreateThread(function()
-            TaskPlayAnim(ped, "mp_am_hold_up", "holdup_victim_20s", 8.0, -8.0, -1, 49, 0, false, false, false)
-            PlayPain(ped, 6, 0)
-        end)
-        Wait(math.random(200,500))
-    end
-
-    -- Cancel progress bar if: Ped dies during robbery, car gets too far away
-    CreateThread(function()
-        while isCarjacking do
-            local distance = #(GetEntityCoords(PlayerPedId()) - GetEntityCoords(target))
-            if IsPedDeadOrDying(target) or distance > 7.5 then
-                TriggerEvent("progressbar:client:cancel")
-            end
-            Wait(100)
+    if Config.CarJackEnable then
```

Comment:
This check is not required since you have already checked if the config setting is true before running this function.

Reply:
I originally had this check at line 100 and figured I didn't have to check on the function as the function is only being called from this piece of code.

Then I thought about it and figured maybe I should put the check inside the function just in-case someone decided to use the function for something else at least the check will be inside the function too.

Remember, functions are reused we want the check inside the function.

I left the check in place at line 100 because the extra work doesn't need to be done if the feature is disabled.

Reply:
The problem with that is if someone decided to use the function for something else, then it's likely that they actually want to use the function. So the control of the config option should reside in the resource that is calling it.

Reply:
We can do one or the other, either put it inside the function or put it everywhere we call it. No need for both.

Reply:
If we put it inside the function, we can also just write it like this on the first line of the function and save an extra indentation:
```lua
if not Config.CarJackEnable then return end
```

Reply:
I let the extra code run as it should be disabled in the function.

Reply:
Alright yeah, that extra code a downside. Let's just add it to both as you had done, I guess it covers all bases, my bad.

Reply:
Done and done and tested and verified on my end.

Want to test it and give it a go?